adafruit_rockblock.py

```diff
@@ -199,6 +206,229 @@ def model(self):
             return resp[1].strip().decode()
         return None
 
-    def _transfer_buffer(self):
-        """Copy out buffer to in buffer to simulate receiving a message."""
-        self._uart_xfer("+SBDTC")
+    @property
+    def serial_number(self):
+        """Modem's serial number, also known as the modem's IMEI.
+
+        Returns
+        string
+        """
+        resp = self._uart_xfer("+CGSN")
+        if resp[-1].strip().decode() == "OK":
+            return resp[1].strip().decode()
+        return None
+
+    @property
+    def signal_quality(self):
+        """Signal Quality also known as the Received Signal Strength Indicator (RSSI).
+
+        Values returned are 0 to 5, where 0 is no signal (0 bars) and 5 is strong signal (5 bars).
+
+        Important note: signal strength may not be fully accurate, so waiting for
+        high signal strength prior to sending a message isn't always recommended.
+        For details see https://docs.rockblock.rock7.com/docs/checking-the-signal-strength
+
+        Returns:
+        int
+        """
+        resp = self._uart_xfer("+CSQ")
+        if resp[-1].strip().decode() == "OK":
+            return int(resp[1].strip().decode().split(":")[1])
+        return None
+
+    @property
+    def revision(self):
+        """Modem's internal component firmware revisions.
+
+        For example: Call Processor Version, Modem DSP Version, DBB Version (ASIC),
+        RFA VersionSRFA2), NVM Version, Hardware Version, BOOT Version
+
+        Returns a tuple:
+        (string, string, string, string, string, string, string)
+        """
+        resp = self._uart_xfer("+CGMR")
+        if resp[-1].strip().decode() == "OK":
+            lines = []
+            for x in range(1, len(resp) - 2):
+                line = resp[x]
+                if line != b"\r\n":
+                    lines.append(line.decode().strip())
+            return tuple(lines)
+        return (None,) * 7
+
+    @property
+    def ring_alert(self):
+        """The current ring indication mode.
+
+        False means Ring Alerts are disabled, and True means Ring Alerts are enabled.
+
+        When SBD ring indication is enabled, the ISU asserts the RI line and issues
+        the unsolicited result code SBDRING when an SBD ring alert is received.
+        (Note: the network can only send ring alerts to the ISU after it has registered).
+
+        Returns:
+        bool
+        """
+        resp = self._uart_xfer("+SBDMTA?")
+        if resp[-1].strip().decode() == "OK":
+            return bool(int(resp[1].strip().decode().split(":")[1]))
+        return None
+
+    @ring_alert.setter
+    def ring_alert(self, value=1):
```

Comment:
Is a default value needed here? Can't really call a setter without passing in something.

Reply:
Ah, you're right, a default value is not needed. I've changed the code.